bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32Handler.java

```diff
@@ -0,0 +1,248 @@
+        }
+    }
+
+    private void checkConnectionAndUpdate() {
+        if (this.thing.getStatus() != ThingStatus.ONLINE) {
+            try {
+                if (httpClient.GET(config.getUri()).getStatus() == 200) {
+                    updateStatus(ThingStatus.ONLINE);
+                    // rerun immediately to update state
+                    checkConnectionAndUpdate();
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "WlanThermo not found under given address.");
+                }
+            } catch (URISyntaxException | ExecutionException | TimeoutException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Could not connect to WlanThermo at " + config.getIpAddress() + ": " + e.getMessage());
+            } catch (InterruptedException e) {
+                logger.debug("Connection check interrupted. {}", e.getMessage());
+            }
+        } else {
+            try {
+                // Update objects with data from device
+                String json = httpClient.GET(config.getUri("/data")).getContentAsString();
+                data = gson.fromJson(json, Data.class);
+                logger.debug("Received at /data: {}", json);
+                json = httpClient.GET(config.getUri("/settings")).getContentAsString();
+                settings = gson.fromJson(json, Settings.class);
+                logger.debug("Received at /settings: {}", json);
+
+                if (data == null || settings == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Failed to parse Data and/or Settings values!");
+                    return;
+                }
+
+                // Update Channels if required
+                Map<String, String> properties = editProperties();
+                Boolean pmEnabled = settings.getFeatures().getBluetooth();
+                int pmChannels = pmEnabled ? data.getPitmaster().getPm().size() : 0;
+                int tempChannels = data.getChannel().size();
+
+                // Update properties
+                properties.put(WlanThermoBindingConstants.PROPERTY_MODEL, settings.getDevice().getDevice());
+                properties.put(WlanThermoBindingConstants.PROPERTY_SERIAL, settings.getDevice().getSerial());
+                properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_BT_ENABLED,
+                        settings.getFeatures().getBluetooth().toString());
+                properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_PM_ENABLED, pmEnabled.toString());
+                properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_TEMP_CHANNELS, String.valueOf(tempChannels));
+                properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_PM_CHANNELS, String.valueOf(pmChannels));
+                updateProperties(properties);
```

Comment:
Why are you changing the properties every time? Thing properties should generally be considered immutable characteristics of the device so you should only need to update it when something changes.

Reply:
The number of temperature and pitmaster channels might change during operation as the user attaches bluetooth sensors. 
On the one hand, I do understand that properties should usually be immutable, but on the other hand I don't think that having the number of attached sensors is worth an own channel. 
I've changed the code so that only the channel number properties are updated, is that okay for you?

Reply:
It wasn't really what I had in mind but it doesn't really sound like you can avoid updating the properties. So I guess this is fine.